From b9b26d466efa44907b865f6d41802b9de687dbbe Mon Sep 17 00:00:00 2001
From: Arthur Heymans <arthur@aheymans.xyz>
Date: Fri, 15 Oct 2021 18:06:15 +0200
Subject: [PATCH 2/3] arm/mach-aspeed: Add spansion S25FL256L

Signed-off-by: Arthur Heymans <arthur@aheymans.xyz>

diff --git a/arch/arm/mach-aspeed/flash.c b/arch/arm/mach-aspeed/flash.c
index b8a438df75..0c922e6813 100644
--- a/arch/arm/mach-aspeed/flash.c
+++ b/arch/arm/mach-aspeed/flash.c
@@ -69,6 +69,7 @@ flash_info_t flash_info[CONFIG_SYS_MAX_FLASH_BANKS];		/* FLASH chips info */
 #define S25FL064A		0x160201
 #define S25FL128P		0x182001
 #define S25FL256S		0x190201
+#define S25FL256L		0x196001
 #define W25X16			0x1530ef
 #define W25X64			0x1730ef
 #define W25Q64BV		0x1740ef
@@ -976,6 +977,20 @@ static ulong flash_get_size (ulong base, flash_info_t *info)
 #endif
 			break;
 
+		case S25FL256L:
+			info->sector_count = 512;
+			info->size = 0x2000000;
+			info->address32 = 1;
+			erase_region_size  = 0x10000;
+			info->readcmd = 0x0b;
+			info->dualport = 0;
+			info->dummybyte = 1;
+			info->buffersize = 256;
+			WriteClk = 50;
+			EraseClk = 20;
+			ReadClk  = 50;
+			break;
+
 		case MX25L25635E:
 			info->sector_count = 256;
 			info->size = 0x1000000;
@@ -1348,7 +1363,7 @@ static ulong flash_get_size (ulong base, flash_info_t *info)
 		*((volatile ulong*) (info->reg_base + 0x4)) = reg;
 
 		/* set flash chips to 32bits addressing mode */
-		if ((info->flash_id & 0xFF) == 0x01)	/* Spansion */
+		if ((info->flash_id & 0xFF) == 0x01 && 0)	/* Spansion */
 			enable4b_spansion(info);
 		else if ((info->flash_id & 0xFF) == 0x20)	/* Numonyx */
 			enable4b_numonyx(info);
-- 
2.31.1

