From 8353c21bcfa39b0cc2daf37303843447e5891489 Mon Sep 17 00:00:00 2001
From: Arthur Heymans <arthur@aheymans.xyz>
Date: Fri, 15 Oct 2021 18:06:40 +0200
Subject: [PATCH 3/3] board/ast-g5: Enable the ethernet PHY's

Signed-off-by: Arthur Heymans <arthur@aheymans.xyz>

diff --git a/board/aspeed/ast-g5/ast-g5.c b/board/aspeed/ast-g5/ast-g5.c
index e67a4bf8b2..686c27b486 100644
--- a/board/aspeed/ast-g5/ast-g5.c
+++ b/board/aspeed/ast-g5/ast-g5.c
@@ -21,6 +21,14 @@ int board_init(void)
 	gd->bd->bi_boot_params = CONFIG_SYS_SDRAM_BASE + 0x100;
 	gd->flags = 0;
 
+	unsigned int gpio_efgh_dir = readl(AST_GPIO_BASE + 0x24);
+	gpio_efgh_dir |= (1 << 7) | (1 << 18); // GPIO E7, G2 (MAC RST) to output
+	writel(gpio_efgh_dir, AST_GPIO_BASE + 0x24);
+
+	unsigned int gpio_efgh_data = readl(AST_GPIO_BASE + 0x20);
+	gpio_efgh_data |= (1 << 7) | (1 << 18); // GPIO E7, G2 (MAC RST) to high
+	writel(gpio_efgh_data, AST_GPIO_BASE + 0x20);
+
 	return 0;
 }
 
-- 
2.31.1

